src/axes.jl

```diff
@@ -366,18 +366,20 @@ end
 # -------------------------------------------------------------------------
 
 # push the limits out slightly
-function widen(lmin, lmax)
-    span = lmax - lmin
+function widen(lmin, lmax, scale)
+    sf = scalefunc(scale)
+    isf = invscalefunc(scale)
+    span = sf(lmax) - sf(lmin)
     # eps = max(1e-16, min(1e-2span, 1e-10))
     eps = max(1e-16, 0.03span)
-    lmin-eps, lmax+eps
+    isf(lmin-eps), isf(lmax+eps)
 end
 
 # figure out if widening is a good idea.  if there's a scale set it's too tricky,
 # so lazy out and don't widen
 function default_should_widen(axis::Axis)
     should_widen = false
-    if axis[:scale] == :identity && !is_2tuple(axis[:lims])
+    if (!(axis[:scale] in _logScales && axis[:extrema].emin <= 0)) && !is_2tuple(axis[:lims])
```

Comment:
No it doesn't, it means it would allow it for log scales as well, but with a provision for not doing it if min is 0 (which it never should be in the first place if the axis is log).
`widen` is the function that creates a little space around the points, so that the limit of the plotted objects don't intersect the axis limits.